src/galaxy/query_builder/builder.py

```diff
@@ -204,23 +204,68 @@ def create_users_contributions_query(params, changeset_query):
     SELECT user_id,
         username,
         total_buildings,
-        public.tasks_per_user(user_id,
-            '{project_ids}',
-            '{from_timestamp}',
-            '{to_timestamp}',
-            'MAPPED') AS mapped_tasks,
-        public.tasks_per_user(user_id,
-            '{project_ids}',
-            '{from_timestamp}',
-            '{to_timestamp}',
-            'VALIDATED') AS validated_tasks,
         public.editors_per_user(user_id,
-            '{from_timestamp}',
-            '{to_timestamp}') AS editors
+        '{from_timestamp}',
+        '{to_timestamp}') AS editors
     FROM T3;
     """
     return query
 
+def create_user_time_spent_mapping_and_validating_query(params):
+    project_ids = ",".join([str(p) for p in params.project_ids])
```

Comment:
We have several conditions for Mapathon Reports , For eg: Project ID can be empty it is optional field user may can only supply hashtag or only project id , This generator will work for those cases where project id is supplied but will not work when hashtag is supplied , Here is the sample query output of this function 
```
        SELECT DISTINCT th.user_id,
            (
                SELECT COUNT(task_id)
                FROM public.task_history
                WHERE user_id = th.user_id
                AND action_text = 'MAPPED'
                AND action_date BETWEEN '2021-09-23 18:30:00' AND '2021-09-24 18:15:00'
                AND project_id IN ()
            ) AS mapped_task_count,
            (
                SELECT COUNT(task_id)
                FROM public.task_history
                WHERE user_id = th.user_id
                AND action_text = 'VALIDATED'
                AND action_date BETWEEN '2021-09-23 18:30:00' AND '2021-09-24 18:15:00'
                AND project_id IN ()
            ) AS validated_task_count
        FROM public.task_history th
        WHERE th.action_text = 'MAPPED' OR th.action_text = 'VALIDATED'
```